ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -200,36 +214,159 @@ public static SourceBuilder instance() {
     );
   }
 
-  private static PhysicalSchema getPhysicalSchemaWithKeyAndPseudoCols(
+  private static PhysicalSchema getPhysicalSchemaWithPseudoColumnsToMaterialize(
       final SourceStep<?> streamSource) {
 
     final boolean windowed = streamSource instanceof WindowedTableSource;
 
     final FormatInfo formatInfo = streamSource.getFormats().getKeyFormat();
     final SerdeFeatures serdeFeatures = streamSource.getFormats().getKeyFeatures();
+
     final KeyFormat keyFormat = windowed
         ? KeyFormat.windowed(
-            formatInfo, serdeFeatures, ((WindowedTableSource) streamSource).getWindowInfo())
+        formatInfo, serdeFeatures, ((WindowedTableSource) streamSource).getWindowInfo())
         : KeyFormat.nonWindowed(formatInfo, serdeFeatures);
 
-    final Formats formats = of(keyFormat, streamSource.getFormats().getValueFormat());
+    final Formats format = ((TableSource) streamSource).getStateStoreVersion();
```

Comment:
This makes windowed tests fail, but once we revert `WindowedTableSource` to be the V1 version, this won't be an issue.